File: VB/MailMergeSample/StreamCopyHelper.vb
Imports System.IO

Public Module StreamCopyHelper

    Public Sub Copy(ByVal src As Stream, ByVal dst As Stream)
        Const bufferSize As Integer = 32768
        Dim buffer As Byte() = New Byte(32767) {}
        Dim bytesRead As Integer = 0
        Do
            bytesRead = src.Read(buffer, 0, bufferSize)
            dst.Write(buffer, 0, bytesRead)
        Loop While bytesRead = bufferSize
    End Sub
End Module

File: CS/MailMergeSample/StreamCopyHelper.cs
using System.IO;

public static class StreamCopyHelper {
    public static void Copy(Stream src, Stream dst) {
        const int bufferSize = 32768;
        byte[] buffer = new byte[bufferSize];
        int bytesRead = 0;
        do {
            bytesRead = src.Read(buffer, 0, bufferSize);
            dst.Write(buffer, 0, bytesRead);

        }
        while (bytesRead == bufferSize);
    }
}